Assets/Scripts/Gameplay/GameplayObjects/Character/ServerCharacter.cs

```diff
@@ -1,30 +1,109 @@
+using System;
 using System.Collections;
-using Unity.Multiplayer.Samples.BossRoom.Actions;
+using Unity.BossRoom.Gameplay.Actions;
+using Unity.BossRoom.Gameplay.Configuration;
+using Unity.BossRoom.Gameplay.GameplayObjects.Character.AI;
+using Unity.Multiplayer.Samples.BossRoom;
 using Unity.Netcode;
 using UnityEngine;
+using Action = Unity.BossRoom.Gameplay.Actions.Action;
 
-namespace Unity.Multiplayer.Samples.BossRoom.Server
+namespace Unity.BossRoom.Gameplay.GameplayObjects.Character
 {
-    public class ServerCharacter : NetworkBehaviour
+    /// <summary>
+    /// Contains all NetworkVariables, RPCs and server-side logic of a character.
+    /// </summary>
+    [RequireComponent(typeof(NetworkHealthState),
+        typeof(NetworkLifeState),
+        typeof(NetworkAvatarGuidState))]
+    public class ServerCharacter : NetworkBehaviour, ITargetable
     {
         [SerializeField]
-        NetworkCharacterState m_NetworkCharacterState;
+        ClientCharacterVisualization m_ClientVisualization;
```

Comment:
nit: I usually try to have all my public and serialize field fields at the top, followed by private fields followed by methods. Helps with reading public API from the top when writing other code that uses them.

Reply:
same with other classes

Reply:
cleaned up the ones I managed to find, if more are found later - we can do them in another PR. Maybe we can just run code cleanup on the project with a setting to fix the order of declaration of fields